subsys/bluetooth/controller/hci/hci.c

```diff
@@ -159,19 +172,115 @@ static void reset(struct net_buf *buf, struct net_buf **evt)
 {
 	struct bt_hci_evt_cc_status *ccst;
 
-	ll_reset();
-
 #if CONFIG_BLUETOOTH_CONTROLLER_DUP_FILTER_LEN > 0
 	dup_count = -1;
 #endif
 	/* reset event masks */
 	event_mask = DEFAULT_EVENT_MASK;
 	le_event_mask = DEFAULT_LE_EVENT_MASK;
 
+	if (buf) {
+
+		ll_reset();
+
+		ccst = cmd_complete(evt, sizeof(*ccst));
+		ccst->status = 0x00;
+	}
+#if defined(CONFIG_BLUETOOTH_CONTROLLER_TO_HOST_FC)
+	hci_hbuf_total = 0;
+	hci_hbuf_sent = 0;
+	hci_hbuf_acked = 0;
+	conn_count = 0;
+	atomic_set_bit(&hci_state_mask, HCI_STATE_BIT_RESET);
+printk("reset signal\n");
```

Comment:
This doesn't seem to be correctly indented - also shouldn't you be using BT_DBG?